code/game/machinery/vending.dm

```diff
@@ -3827,3 +3827,117 @@ var/global/num_vending_terminals = 1
+	//The vending machine can have a mouse inside of it! If it does, it has a chance to eject it on each vend.
+	var/hasmouse = 0
+	var/chanceofhavingmouse = 35
+	var/chanceofejectingmouse = 10
+	premium = list(
+		/obj/item/weapon/reagent_containers/food/snacks/sausage/dan = 3,
+		)
+	prices = list(
+		/obj/item/weapon/reagent_containers/food/snacks/meat/animal/dan = 10,
+		/obj/item/weapon/reagent_containers/food/snacks/sausage/dan = 15,
+		/obj/item/weapon/reagent_containers/food/snacks/meat/human = 15,
+		/obj/item/weapon/reagent_containers/food/snacks/meat/carpmeat/imitation = 15,
+		)
+	pack = /obj/structure/vendomatpack/meat
+
+/obj/machinery/vending/meat/New()
+	..()
+	//Chance of a mouse inside of the vending machine
+	if(prob(chanceofhavingmouse))
+		hasmouse = 1
+
+	//Dan isn't really consistent with his new factories. Random amounts of meats are included.
+	//This all goes into New() because rand() can't be called in an object definition.
+	for(var/i = 1 to rand(6,8))
+		add_more_meat()
+
+	contraband = list(
+		/obj/item/weapon/reagent_containers/food/snacks/monkeycube/cowcube = 1,
+		/obj/item/weapon/reagent_containers/food/snacks/meat/human = rand(0,2),
+		/obj/item/weapon/reagent_containers/food/snacks/meat/carpmeat/imitation = rand(0,2)
+		)
+	switch(rand(1,3))
+		if(3)
+			contraband[/obj/item/weapon/reagent_containers/food/snacks/meat/roach/big] = 1
+		else
+			contraband[/obj/item/weapon/reagent_containers/food/snacks/meat/roach] = rand(1,2)
+	src.build_inventory(contraband, 1)
+
+/obj/machinery/vending/meat/proc/add_more_meat()
+	//More meat. More of the same entry. MORE. MORE!!
+	products = list(
+		/obj/item/weapon/reagent_containers/food/snacks/meat/animal/dan = rand(1,6),
+		)
+	src.build_inventory(products)
+
+/obj/machinery/vending/meat/update_vicon()
+	//Override the usual function so we can run special mouse codes
+	if(stat & (BROKEN))
+		//If the mouse is still inside, it isn't anymore... rip
+		if(hasmouse)
+			hasmouse = 0
+			visible_message("\The [src.name] makes a sickening splatter sound.", "You hear a splat.")
+			playsound(loc, 'sound/effects/splat.ogg', 50, 1)
+			src.icon_state = "[initial(icon_state)]-broken"
+		else
+			src.icon_state = "[initial(icon_state)]-brokennomouse"
+	else if (stat & (NOPOWER|FORCEDISABLE))
+		if(hasmouse)
+			src.icon_state = "[initial(icon_state)]-off"
+		else
+			src.icon_state = "[initial(icon_state)]-offnomouse"
+	else
+		src.icon_state = "[initial(icon_state)]"
+
+/obj/machinery/vending/meat/vend(datum/data/vending_product/R, mob/user, by_voucher = 0)
+	..()
+	if(hasmouse && prob(chanceofejectingmouse))
+		hasmouse = 0
+		visible_message("\The [src.name] makes a squeaking sound as a mouse pops out of the slot!", "You hear a squeak.")
+		playsound(loc, 'sound/effects/mousesqueek.ogg', 50, 1)
+		spawn(vend_delay)
+			new /mob/living/simple_animal/mouse/common/dan(get_turf(src))
+
+/obj/machinery/vending/meat/crowbarDestroy(mob/user, obj/item/tool/crowbar/C)
+	..()
+	if(hasmouse)
```

Comment:
instead of pasting the code, it would be more maintainable to just make a mouse ejection proc; also lets admins proccall to eject mouse, which is fun